web/views.py

```diff
@@ -299,6 +300,35 @@ def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
         return context
 
 
+class ParticipantEmailUnsubscribeView(generic.View):
+    template_name = "web/unsubscribe.html"
+
+    def get(self, request, username, *args, **kwargs):
+        return render(request, self.template_name, {"username": username})
```

Comment:
So with this GET request, the user would see this page if they go directly to the `account/<username>` link (with no token)? Or even just `account/`? And then the user sees the 'attempting to unsubscribe' page (web/templates/web/unsubscribe.html)? 
Does this use the session cookie for auth? 
What's the use case for this GET request? Are we handling the case when the user clicks the unsubscribe-all link in the email, but the token has expired, so we're falling back to login for authentication?

Reply:
I'll add a step-by-step rundown of what happens when a user clicks the unsubscribe link to the PR description above.

Reply:
What I could do instead, which might be cleaner, would be to simply redirect to POST in the view.

Reply:
Thanks very much for explaining all of this here and in the PR description. All makes sense now!